src/LogicTasks/Semantics/Fill.hs

```diff
@@ -45,11 +54,11 @@ genFillInst FillConfig{..} = do
 
     formula <- case formulaConfig of
       (FormulaArbitrary syntaxTreeConfig) ->
-        InstArbitrary <$> genSynTree syntaxTreeConfig `suchThat` \t -> withRatio percentTrueEntries' t
+        InstArbitrary <$> genSynTree syntaxTreeConfig `suchThat` \t -> withRatio percentTrueEntries' t && synTreeDependsOnAllAtomics t
       (FormulaCnf cnfCfg) ->
-        tryGen (InstCnf <$> genCnf' cnfCfg) 100 $ withRatio percentTrueEntries'
+        tryGen (InstCnf <$> genCnf' cnfCfg) 100 $ \f -> withRatio percentTrueEntries' f && formulaDependsOnAllAtoms f
       (FormulaDnf dnfCfg) ->
-        tryGen (InstDnf <$> genDnf' dnfCfg) 100 $ withRatio percentTrueEntries'
+        tryGen (InstDnf <$> genDnf' dnfCfg) 100 $ \f -> withRatio percentTrueEntries' f && formulaDependsOnAllAtoms f
```

Comment:
Maybe even here as well, so I'm not 100% certain what the interaction with `tryGen` is. (See also #106.)